cmd/state-tray/internal/menu/projects.go

```diff
@@ -0,0 +1,83 @@
+package menu
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ActiveState/cli/cmd/state-tray/internal/open"
+	"github.com/ActiveState/cli/internal/graph"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/getlantern/systray"
+)
+
+type LocalProjectsUpdater struct {
+	menuItem *systray.MenuItem
+	items    []*localProjectsMenuItem
+}
+
+type localProjectsMenuItem struct {
+	menuItem *systray.MenuItem
+	project  *graph.Project
+	done     <-chan struct{}
+	cancel   context.CancelFunc
+}
+
+func NewLocalProjectsUpdater(menuItem *systray.MenuItem) *LocalProjectsUpdater {
+	return &LocalProjectsUpdater{menuItem, []*localProjectsMenuItem{}}
+}
+
+func (u *LocalProjectsUpdater) Update(projects []*graph.Project) {
+	for _, item := range u.items {
+		item.remove()
+	}
+
+	u.items = make([]*localProjectsMenuItem, len(projects))
+	for i, project := range projects {
+		item := newLocalProjectMenuItem(project, u.menuItem)
+		u.items[i] = item
```

Comment:
Nit: little awkward to depend on the index like this. Why not just use `append()`?

Reply:
Because I know the length of the slice. https://github.com/ActiveState/cli/pull/1293#discussion_r603638661

Reply:
That doesn't mean you can't use `append()`, to my knowledge?